spike_main/spike.cc

```diff
@@ -123,31 +123,48 @@ bool sort_mem_region(const mem_cfg_t &a, const mem_cfg_t &b)
     return (a.base < b.base);
 }
 
-void merge_overlapping_memory_regions(std::vector<mem_cfg_t> &mems)
+static bool check_mem_overlap(const mem_cfg_t& L, const mem_cfg_t& R)
 {
-  // check the user specified memory regions and merge the overlapping or
-  // eliminate the containing parts
-  assert(!mems.empty());
+  reg_t L_start = L.base;
+  reg_t L_end   = L.base + L.size - 1;
+
+  reg_t R_start = R.base;
+  reg_t R_end   = R.base + R.size - 1;
+
+  return std::max(L_start, R_start) <= std::min(L_end, R_end);
```

Comment:
Minor stylistic thing, but I'd just use `L.base` here instead of declaring and using `L_start`.